### PR TITLE
Babel interopRequireWildcard helper reexports

### PR DESCRIPTION
**Pull request by guybedford**
This adds support for reexport patterns based on `var _x = _interopRequireWildcard(require('x'))`.

We specifically detect the `"_interopRequireWildcard"` identifier, so like TypeScript helper reexports this will not survive minification, but for npm packages the detection goal of supporting transpiled ES modules is still met despite this potential issue.

Fixes https://github.com/guybedford/cjs-module-lexer/issues/52

/cc @nicolo-ribaudo

### CODE REVIEW

File: lexer.js [hunk 1]
```diff
@@ -97,7 +97,18 @@ function parseSource (cjsSource) {
           lastTokenPos = pos;
           continue;
         case 95/*_*/:
-          if (source.startsWith('_export', pos + 1) && (keywordStart(pos) || source.charCodeAt(pos - 1) === 46/*.*/)) {
+          if (source.startsWith('interopRequireWildcard', pos + 1) && (keywordStart(pos) || source.charCodeAt(pos - 1) === 46/*.*/)) {
+            const startPos = pos;
+            pos += 23;
+            if (source.charCodeAt(pos) === 40/*(*/) {
```
Comment (nicolo-ribaudo) [inline lexer.js:103]: Shouldn't we `break` when the char is not `(`?

**Comment by nicolo-ribaudo** [comment]
I feel like detecting `interopRequireWildcard` specifically is fragile, but:
1. I can't think of a better pattern
2. Most modules on `npm` aren't minified anyway

File: lexer.js (hunk 1 above)
Comment (guybedford) [reply lexer.js:103]: The loop is designed to support fall-through as this is the main parse loop that iterates each character.

**Comment by guybedford** [comment]
@nicolo-ribaudo how reliable is the `_interopRequireWildcard` name here (with the `_`)? I know it's fragile, it would just be nice to get a sense of how fragile. Should we support without the leading `_`, and possibly with `$n`?

**Comment by nicolo-ribaudo** [comment]
Babel will never inject it without the leading `_`.

`$n` could happen when you compile a file which has already been compiled (thus it already contains an `_interopRequireWildcard` function). However, when people compile re-compile files it's almost always because they are bundling and compiling their dependencies (and in this case Babel's exports wouldn't be the final bundle exports anyway).

**Comment by guybedford** [comment]
Thanks for confirming!

**Pull request merged**